app/main/register-sentry-state-listener.ts

```diff
@@ -1,17 +1,22 @@
 import { ipcMain } from 'electron';
-import * as Sentry from '@sentry/electron/dist/main';
+import * as Sentry from '@sentry/electron';
+import ElectronStore from 'electron-store';
 
-let diagnosticsEnabled = false;
+let diagnosticsEnabled = true;
```

Comment:
Why do we now default to enabling diagnostics? This should always be opt in.

Reply:
Yes same comment here, it's better to always default to false. It also makes our intent clear if someone else is reading the code.

Reply:
@kyranjamie We still need it so we return null when the user changes the setting. We can remove it as we give the warning that changes will only update when the app restarts though.
@beguene we can't set it to false by default as we only initialize sentry if the setting is true, so it has to be true initially and we switch it to false when the user does.

Reply:
I don't really understand this. 

If a completely new user installs and opens the wallet, this function is ran immediately, and `diagnosticsEnabled` default to true. So we enable diagnostics, before the user has made a decision?

Reply:
the function `initializeSentry` checks for the setting value in store before calling sentry.init
<img width="632" alt="image" src="https://user-images.githubusercontent.com/97111756/155340987-b49556cd-9dc5-4566-b32a-39eb0d7d3e65.png">
if the setting is true, we wanna initialize and allow sending events, thus diagnosticsEnabled is set to true.
If the user changes the setting to false, diagnosticsEnabled will be false and beforeSend returns null, we still ask users to restart the app as sentry (after initialization) can still attempt to call its server.

Reply:
happy to jump on a call to discuss this further.